src/pkg/cached/base_manager.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cached
+
+import (
+	"context"
+	"time"
+
+	"github.com/goharbor/harbor/src/lib/cache"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/orm"
+)
+
+// innerCache is the default cache client.
+var innerCache cache.Cache = &cacheClient{}
+
+// cacheClient is a interceptor for cache.Default, in order to implement specific
+// case for cache layer.
+type cacheClient struct{}
+
+func (*cacheClient) Contains(ctx context.Context, key string) bool {
+	return cache.Default().Contains(ctx, key)
+}
+
+func (*cacheClient) Delete(ctx context.Context, key string) error {
+	return cache.Default().Delete(ctx, key)
+}
+
+func (*cacheClient) Fetch(ctx context.Context, key string, value interface{}) error {
+	return cache.Default().Fetch(ctx, key, value)
+}
+
+func (*cacheClient) Ping(ctx context.Context) error {
+	return cache.Default().Ping(ctx)
+}
+
+func (*cacheClient) Save(ctx context.Context, key string, value interface{}, expiration ...time.Duration) error {
+	// intercept here
+	// it should ignore save cache if this request is wrapped by orm.Transaction,
```

Comment:
please add comment for the meaning of commitedKey.

Reply:
updated